### ISSUE TITLE
CA1801: ReviewUnusedParameters should not trigger on properties

### ISSUE THREAD

**Issue by ivanbasov** [dotnet/roslyn-analyzers#1466]
**Example 1.** Empty setter:
`int Property1 { get { return 0;} set { }}`
Expected:
No warning
Acutal:
CA1801 for 'value'

**Example 2.** Unused indexer
```
    public int this[int i]
    {
        get { return 0; }
        set { Console.WriteLine(value); }
    }
```
Expected:
No warning
Acutal:
CA1801 for 'i'

**Example 3.** There is another rule suggesting to avoid multiple indexers. Need to check that CA1801 is not triggered.

Currently, we have to skip fixing those cases because they lead to completely invalid code. Need to consider better options.

**Comment by 333fred** [comment]
Arguably, we should trigger a different warning here. Something like: you're not using the backing store/indexer property, consider using them for something. If the user actually doesn't want to use them, they can suppress it.

**Comment by mavasani** [comment]
Verified no CA1801 on none of the cases in the description with latest FxCop analyzers package
